core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -126,13 +125,48 @@ class KafkaApis(val requestChannel: RequestChannel,
     info("Shutdown complete.")
   }
 
+  private def validateForwardRequest(request: RequestChannel.Request): Unit = {
+    if (!config.forwardingEnabled) {
+      throw new InvalidRequestException(s"The handling of forwarding request $request is not enabled on this broker.")
+    } else if (!request.context.fromPrivilegedListener ||
```

Comment:
This begs the question whether the api should even be advertised from non-privileged listeners if users cannot access it. I am thinking we can make this case similar to the behavior if forwarding is not enabled.  Here we can use this logic:
```
if (!config.forwardingEnabled || !request.context.fromPrivilegedListener) {
  closeConnection(request, util.Collections.emptyMap())
} else if (!authorize(request.envelopeContext.get.brokerContext, CLUSTER_ACTION, CLUSTER, CLUSTER_NAME)) {
  throw new ClusterAuthorizationException(s"Envelope request $request is not authorized")
} 
```
Similarly, we can change the check in `ApiVersion.apiVersionsResponse` so that it skips the ENVELOPE API if the request is not from a privileged listener.